### PR TITLE
split handlers into their own packages

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

All handlers existed in one single package. This PR separates them into their own packages to enable quicker tests and make the codebase a bit more readable (still way to go)

- make tests faster
- fix broken tests

**Related issue(s)**

### CODE REVIEW

File: components/eventing-controller/pkg/subscriptionmanager/beb/beb.go [hunk 1]
```diff
@@ -6,7 +6,9 @@ import (
 	"strings"
 	"time"
 
+	beb2 "github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/beb"
```
Comment (mfaizanse) [inline components/eventing-controller/pkg/subscriptionmanager/beb/beb.go:9]: (nit) can we use better name than `beb2`?

File: components/eventing-controller/pkg/subscriptionmanager/jetstream/jetstream.go [hunk 2]
```diff
@@ -5,6 +5,8 @@ import (
 	"fmt"
 
 	pkgmetrics "github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/metrics"
+	jetstream2 "github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/nats/jetstream"
```
Comment (mfaizanse) [inline components/eventing-controller/pkg/subscriptionmanager/jetstream/jetstream.go:8]: (nit) can we use better name than `jetstream2`?

File: components/eventing-controller/controllers/subscription/beb/reconciler.go [hunk 3]
```diff
@@ -9,7 +9,9 @@ import (
 	"strings"
 	"time"
 
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/beb"
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/sink"
+	utils2 "github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/utils"
```
Comment (mfaizanse) [inline components/eventing-controller/controllers/subscription/beb/reconciler.go:14]: (nit) can we use better name than `utils2`?

File: components/eventing-controller/controllers/subscription/jetstream/reconciler.go [hunk 4]
```diff
@@ -20,8 +20,9 @@ import (
 	"github.com/kyma-project/kyma/components/eventing-controller/controllers/events"
 	"github.com/kyma-project/kyma/components/eventing-controller/logger"
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/env"
-	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers"
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/eventtype"
+	nats3 "github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/nats"
```
Comment (mfaizanse) [inline components/eventing-controller/controllers/subscription/jetstream/reconciler.go:24]: (nit) can we use better name than `nats3`?

File: components/eventing-controller/controllers/subscription/jetstream/reconciler_test.go [hunk 5]
```diff
@@ -19,14 +19,15 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/envtest"
 
 	eventingv1alpha1 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha1"
+	jetstream2 "github.com/kyma-project/kyma/components/eventing-controller/controllers/subscription/jetstream"
```
Comment (mfaizanse) [inline components/eventing-controller/controllers/subscription/jetstream/reconciler_test.go:22]: (nit) can we use better name than `jetstream2`?

File: components/eventing-controller/controllers/subscription/nats/reconciler.go [hunk 6]
```diff
@@ -22,8 +22,9 @@ import (
 	"github.com/kyma-project/kyma/components/eventing-controller/controllers/events"
 	"github.com/kyma-project/kyma/components/eventing-controller/logger"
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/env"
-	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers"
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/eventtype"
+	nats2 "github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/nats"
```
Comment (mfaizanse) [inline components/eventing-controller/controllers/subscription/nats/reconciler.go:26]: (nit) can we use better name than `nats2`?

File: components/eventing-controller/controllers/subscription/nats/reconciler_test.go [hunk 7]
```diff
@@ -26,9 +26,9 @@ import (
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/application/applicationtest"
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/application/fake"
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/env"
-	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers"
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/eventtype"
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/metrics"
+	nats2 "github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/nats/core"
```
Comment (mfaizanse) [inline components/eventing-controller/controllers/subscription/nats/reconciler_test.go:31]: (nit) can we use better name than `nats2`?

File: components/eventing-controller/pkg/handlers/beb/beb.go [hunk 8]
```diff
@@ -17,6 +17,7 @@ import (
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/ems/httpclient"
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/env"
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/eventtype"
+	utils2 "github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/utils"
```
[the remark this answers is not in the document]
Comment (k15r) [reply components/eventing-controller/pkg/handlers/beb/beb.go]: done

File: components/eventing-controller/pkg/handlers/nats/jetstream/jetstream.go [hunk 9]
```diff
@@ -11,6 +11,8 @@ import (
 	"time"
 
 	pkgmetrics "github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/metrics"
+	ecnats "github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/nats"
+	nats2 "github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/nats/core"
```
[the remark this answers is not in the document]
Comment (k15r) [reply components/eventing-controller/pkg/handlers/nats/jetstream/jetstream.go]: done

File: components/eventing-controller/pkg/subscriptionmanager/beb/beb.go (hunk 1 above)
Comment (k15r) [reply components/eventing-controller/pkg/subscriptionmanager/beb/beb.go]: done

File: components/eventing-controller/controllers/subscription/beb/reconciler.go (hunk 3 above)
Comment (k15r) [reply components/eventing-controller/controllers/subscription/beb/reconciler.go]: done

File: components/eventing-controller/controllers/subscription/jetstream/reconciler.go (hunk 4 above)
Comment (k15r) [reply components/eventing-controller/controllers/subscription/jetstream/reconciler.go]: done

File: components/eventing-controller/controllers/subscription/jetstream/reconciler_test.go (hunk 5 above)
Comment (k15r) [reply components/eventing-controller/controllers/subscription/jetstream/reconciler_test.go:23]: done

File: components/eventing-controller/pkg/subscriptionmanager/nats/nats_test.go [hunk 10]
```diff
@@ -7,13 +7,16 @@ import (
 	"time"
 
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/metrics"
+	nats2 "github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/nats"
```
Comment (raypinto) [inline components/eventing-controller/pkg/subscriptionmanager/nats/nats_test.go:10]: (nit) natsutils?

File: components/eventing-controller/pkg/subscriptionmanager/jetstream/jetstream_test.go [hunk 11]
```diff
@@ -6,21 +6,23 @@ import (
 	"testing"
 	"time"
 
-	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/metrics"
-
 	"github.com/nats-io/nats-server/v2/server"
 
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/metrics"
+	nats2 "github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/nats"
+	jetstream2 "github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/nats/jetstream"
```
Comment (raypinto) [inline components/eventing-controller/pkg/subscriptionmanager/jetstream/jetstream_test.go:13]: (nit) jetstreambackend?

File: components/eventing-controller/pkg/handlers/nats/jetstream/jetstream.go [hunk 12]
```diff
@@ -10,22 +10,23 @@ import (
...
-	pkgmetrics "github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/metrics"
-
 	cev2 "github.com/cloudevents/sdk-go/v2"
 	cev2protocol "github.com/cloudevents/sdk-go/v2/protocol"
 	"github.com/nats-io/nats.go"
 	"github.com/pkg/errors"
 	"go.uber.org/zap"
 
+	pkgmetrics "github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/metrics"
+	ecnats "github.com/kyma-project/kyma/components/eventing-controller/pkg/handlers/nats"
```
Comment (raypinto) [inline components/eventing-controller/pkg/handlers/nats/jetstream/jetstream.go:20]: (nit) natsutils?

File: components/eventing-controller/pkg/handlers/nats/core/core.go [hunk 13]
```diff
@@ -37,13 +34,11 @@ const (
 	natsHandlerName = "nats-handler"
 )
 
-type ConnClosedHandler func(conn *nats.Conn)
-
 type NatsBackend interface {
```
Comment (raypinto) [inline components/eventing-controller/pkg/handlers/nats/core/core.go:37]: Backend?

File: components/eventing-controller/controllers/subscription/jetstream/reconciler_test.go [hunk 14]
```diff
@@ -19,15 +19,16 @@ import (
...
 
 	eventingv1alpha1 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha1"
+	subscriptionjetstream "github.com/kyma-project/kyma/components/eventing-controller/controllers/subscription/jetstream"
 	utils "github.com/kyma-project/kyma/components/eventing-controller/controllers/subscription/testing"
 	"github.com/kyma-project/kyma/components/eventing-controller/logger"
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/application/applicationtest"
 	"github.com/kyma-project/kyma/components/eventing-controller/pkg/application/fake"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/backend/eventtype"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/backend/metrics"
+	natsjetstream "github.com/kyma-project/kyma/components/eventing-controller/pkg/backend/nats/jetstream"
```
Comment (raypinto) [inline components/eventing-controller/controllers/subscription/jetstream/reconciler_test.go:29]: (nit) backendjetstream?

File: components/eventing-controller/pkg/ems/httpclient/client.go [hunk 15]
```diff
@@ -27,18 +27,18 @@ type Client struct {
 // NewHTTPClient creates a new client and ensures that the given baseURL ends with a trailing '/'.
 // The trailing '/' is required later for constructing the full URL using a relative path.
 func NewHTTPClient(baseURL string, client *http.Client) (*Client, error) {
-	url, err := url.Parse(baseURL)
+	u, err := url.Parse(baseURL)
```
Comment (raypinto) [inline components/eventing-controller/pkg/ems/httpclient/client.go:30]: (nit) can we have a better variable name

**Review by raypinto — APPROVED** [review]

**Pull request merged**